### PR TITLE
Tidy test script and fix output

### PR DESCRIPTION
**Pull request by pickfire**

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -61,32 +62,32 @@ then
 	fi
 
 	echo "> Creating images"
-	dd if=/dev/zero of=block count=10 bs=1M 2>&1 | sed 's/^/\t/g'
+	dd if=/dev/zero of=block count=10 bs=1M 2>&1 | sed 's/^/\t/'
```
Comment (7heo) [inline test.sh]: This `global` flag is needed for the multi-line outputs.

File: test.sh [hunk 2]
```diff
@@ -61,32 +62,32 @@ then
...
 	echo "> Setting up block as $block"
-	sudo losetup $block block 2>&1 | sed 's/^/\t/g'
+	sudo losetup $block block 2>&1 | sed 's/^/\t/'
 
 	echo "> Formatting '$block' with passphrase '$passphrase'."
-	printf "%s" "$passphrase" | sudo cryptsetup luksFormat -q $block - 2>&1 | sed 's/^/\t/g'
+	printf "%s" "$passphrase" | sudo cryptsetup luksFormat -q $block - 2>&1 | sed 's/^/\t/'
 	echo "> Opening the device '$block' as /dev/mapper/temp-test"
-	printf "%s" "$passphrase" | sudo cryptsetup luksOpen -q $block temp-test - 2>&1 | sed 's/^/\t/g'
+	printf "%s" "$passphrase" | sudo cryptsetup open -q $block temp-test - 2>&1 | sed 's/^/\t/'
```
Comment (7heo) [inline test.sh]: `luksOpen` is replaced by `open --type luks`, not only by `open`.
The `global` `sed(1)` flag is needed for the multi-line outputs.

File: test.sh [hunk 3]
```diff
@@ -33,14 +33,15 @@ usage () {
...
-		"-"*) shift;; # Ignore erroneous flags
+		"-d") flags="-d $flags";;
+		"-y") noconfirm=1;;
+		"-x") clean_all=1;;
+		"-h"|"--help") usage;;
+		"-k") clean="";;
+		"-"*) :;; # Ignore erroneous flags
 		*) break;;
 	esac
+	shift
```
Comment (7heo) [inline test.sh:44]: :+1:

File: test.sh [hunk 4]
```diff
@@ -61,32 +62,32 @@ then
...
-	printf "%s" "$passphrase" | sudo cryptsetup luksOpen -q $block temp-test - 2>&1 | sed 's/^/\t/g'
+	printf "%s" "$passphrase" | sudo cryptsetup open -q $block temp-test - 2>&1 | sed 's/^/\t/'
 	echo "> Creating a filesystem on '/dev/mapper/temp-test'"
 	sudo mkfs.ext2 /dev/mapper/temp-test
 	echo "> Mounting the fs"
 	sudo mount -t ext2 /dev/mapper/temp-test local-mount
 	echo "> Creating proof in the mounted fs"
-	sudo sh -c 'date "+proof:%s" > local-mount/proof'
-	proof=$(cat local-mount/proof)
+	proof=$(date "+proof:%s" | sudo tee local-mount/proof)
```
Comment (7heo) [inline test.sh]: No. This is terribly wrong. We need to write the data first and **then** fetch it from the filesystem, to ensure that it is present. :-1:

File: test.sh [hunk 5]
```diff
@@ -61,32 +62,32 @@ then
...
 	echo "> Proof is: '$proof'"
 	echo "> Unmounting the fs"
 	sudo umount local-mount
 	echo "> Closing the device '/dev/mapper/temp-test'"
-	sudo cryptsetup luksClose temp-test
+	sudo cryptsetup close temp-test
 
 	echo "> Testing nlplug-findfs on $block (passphrase was '$passphrase')"
 	echo "$passphrase" | sudo ./nlplug-findfs -p /sbin/mdev ${flags} -c $block -m 'test-device' /dev/mapper/test-device || retcode=1
+	echo
```
Comment (7heo) [inline test.sh]: Why?

File: test.sh [hunk 6]
```diff
@@ -61,32 +62,32 @@ then
...
 	echo "> Creating proof in the mounted fs"
-	sudo sh -c 'date "+proof:%s" > local-mount/proof'
-	proof=$(cat local-mount/proof)
+	proof=$(date "+proof:%s" | sudo tee local-mount/proof)
 	echo "> Proof is: '$proof'"
 	echo "> Unmounting the fs"
 	sudo umount local-mount
 	echo "> Closing the device '/dev/mapper/temp-test'"
-	sudo cryptsetup luksClose temp-test
+	sudo cryptsetup close temp-test
```
Comment (7heo) [inline test.sh]: Unlike the `open --type luks` syntax, this is the correct one (the man documents it as the command `luksClose` is aliased to). :+1:

**Review by 7heo — CHANGES_REQUESTED** [review]

File: test.sh [hunk 7]
```diff
@@ -61,32 +62,32 @@ then
...
 
 	echo "> Setting up the loop devices"
 	block="$(sudo losetup -f)"
 	echo "> Setting up block as $block"
-	sudo losetup $block block 2>&1 | sed 's/^/\t/g'
+	sudo losetup $block block 2>&1 | sed 's/^/\t/'
 
 	echo "> Formatting '$block' with passphrase '$passphrase'."
-	printf "%s" "$passphrase" | sudo cryptsetup luksFormat -q $block - 2>&1 | sed 's/^/\t/g'
+	printf "%s" "$passphrase" | sudo cryptsetup luksFormat -q $block - 2>&1 | sed 's/^/\t/'
```
Comment (pickfire) [inline test.sh]: @7heo The `global` is required only for repeats in multi line output, substituting `^` would not give us any repeats.

File: test.sh [hunk 8]
```diff
@@ -61,32 +62,32 @@ then
...
 
 	echo "> Creating images"
-	dd if=/dev/zero of=block count=10 bs=1M 2>&1 | sed 's/^/\t/g'
+	dd if=/dev/zero of=block count=10 bs=1M 2>&1 | sed 's/^/\t/'
 
 	echo "> Setting up the loop devices"
 	block="$(sudo losetup -f)"
 	echo "> Setting up block as $block"
-	sudo losetup $block block 2>&1 | sed 's/^/\t/g'
+	sudo losetup $block block 2>&1 | sed 's/^/\t/'
```
Comment (pickfire) [inline test.sh:71]: @7heo The `global` is required only for repeats in multi line output, substituting `^` would not give us any repeats.

File: test.sh (hunk 2 above)
Comment (pickfire) [reply test.sh]: Okay, I will change it with `open --type luks`.

File: test.sh (hunk 4 above)
Comment (pickfire) [reply test.sh]: Ah, I think I understand what you mean.

File: test.sh (hunk 5 above)
Comment (pickfire) [reply test.sh]: To clean up the output.

File: test.sh (hunk 1 above)
Comment (pickfire) [reply test.sh]: @7heo The `global` is required only for repeats in multi line output, substituting ^ would not give us any repeats.

File: test.sh (hunk 5 above)
Comment (7heo) [reply test.sh]: How does adding a line clean the output?

File: test.sh (hunk 1 above)
Comment (7heo) [reply test.sh]: Actually global is required in multi-line outputs, but not for `sed`, which parses lines one by one. So this is correct.

File: test.sh (hunk 8 above)
Comment (7heo) [inline test.sh]: See above.

File: test.sh (hunk 7 above)
Comment (7heo) [inline test.sh]: See above.

**Review by 7heo — CHANGES_REQUESTED** [review]
Could you please squash this commit in the previous one?

**Comment by pickfire** [comment]
How do I do that in github?

**Comment by 7heo** [comment]
Do it locally and issue a `git push -f`

**Comment by 7heo** [comment]
@pickfire also this needs a rebase now.

**Comment by pickfire** [comment]
Updated.

**Pull request closed without merging**